app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -59,4 +71,65 @@ class AllFilesFragment : Fragment(), ItemClickListener {
     override fun onItemClickListener() {
         Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
     }
+
+    private fun showPermissionDialog() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            try {
+                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                intent.addCategory("android.intent.category.DEFAULT")
+                intent.data = Uri.parse(String.format("package:%s", arrayOf<Any>(applicationContext().packageName)))
+                activity?.startActivityForResult(intent, 2000)
+            } catch (e: Exception) {
+                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                activity?.startActivityForResult(intent, 2000)
+            }
+        } else ActivityCompat.requestPermissions(
+            super.requireActivity(),
```

Comment:
do you need to call `super.`  ? what is the difference from just calling `requireActivity()`?